### PR TITLE
EN-40387: Not selecting group by constant result in error

### PR DESCRIPTION
**Pull request by chitang**
QA: To repro:
1. create soql view - select field, ' ' as constant group by field, constant
2. $query=select field -> error

### CODE REVIEW

File: common-pg/src/test/scala/com/socrata/pg/soql/SqlizerBasicTest.scala [hunk 1]
```diff
@@ -395,10 +395,17 @@ class SqlizerBasicTest extends SqlizerTest {
...
     val ParametricSql(Seq(sql), setParams) = sqlize(soql, CaseSensitive)
-    sql should be ("SELECT t1.id,e'stRing',5,(2 * 3) FROM t1 GROUP BY t1.id,2,3,4")
+    sql should be ("SELECT t1.id,e'stRing',5,(2 * 3) FROM t1 GROUP BY t1.id")
+    setParams.length should be (0)
+  }
+
+  test("group by literals all removed") {
+    val soql = "select 'stRing' as a, 5 as b, 2*3 as c group by a, b, c"
+    val ParametricSql(Seq(sql), setParams) = sqlize(soql, CaseSensitive)
+    sql should be ("SELECT e'stRing',5,(2 * 3) FROM t1")
```
Comment (znep) [inline common-pg/src/test/scala/com/socrata/pg/soql/SqlizerBasicTest.scala:408]: Wouldn't this need a `limit 1` to be equiv?

Comment (chitang) [reply common-pg/src/test/scala/com/socrata/pg/soql/SqlizerBasicTest.scala:408]: hm...yes, this edge case wouldn't work.

Comment (chitang) [reply common-pg/src/test/scala/com/socrata/pg/soql/SqlizerBasicTest.scala]: fixed in a different way
